lc0_main.go

```diff
@@ -203,30 +205,32 @@ func createCmdWrapper() *cmdWrapper {
 func (c *cmdWrapper) launch(networkPath string, args []string, input bool) {
 	dir, _ := os.Getwd()
 	c.Cmd = exec.Command(path.Join(dir, "lc0"))
-	c.Cmd.Args = append(c.Cmd.Args, args...)
-	c.Cmd.Args = append(c.Cmd.Args, fmt.Sprintf("--weights=%s", networkPath))
+	// Add the "selfplay" or "uci" part first
+	mode := args[0]
+	c.Cmd.Args = append(c.Cmd.Args, mode)
+	args = args[1:]
 	if *lc0Args != "" {
-		// Strict checking of the --lc0args options to prevent someone
-		// from passing a different visits or batch size for example.
+		log.Println("WARNING: Option --lc0args is for testing, not production use!")
+		log.SetPrefix("TESTING: ")
 		parts := strings.Split(*lc0Args, " ")
-		for _, opt := range parts {
-			words := regexp.MustCompile("[,=().0-9]").Split(opt, -1)
-			for _, word := range words {
-				optOK := false
-				switch word {
-					case "", "--backend", "tf", "cudnn", "opencl", "blas", "cudnn-fp",
-						"multiplexing", "--backend-opts", "backend", "gpu", "verbose",
-						"true", "false", "--parallelism":
-					optOK = true
-				}
-				if !optOK {
-					log.Fatalf("Not accepted --lc0args option: %s", opt)
-				}
+		c.Cmd.Args = append(c.Cmd.Args, parts...)
+	}
+	if *backopts != "" {
+		// Check agains small token blacklist, currently only "random"
+		tokens := regexp.MustCompile("[,=().0-9]").Split(*backopts, -1)
+		for _, token := range tokens {
+			switch token {
+				case "random":
+				log.Fatalf("Not accepted in --backend-opts: %s", token)
 			}
 		}
-
-		c.Cmd.Args = append(c.Cmd.Args, parts...)
+		c.Cmd.Args = append(c.Cmd.Args, fmt.Sprintf("--backend-opts=%s", *backopts))
 	}
+	if *parallel != -1 && mode == "selfplay" {
```

Comment:
if *parallel > 0  (prevent a negative value on the command line)

Reply:
OK